src/manifest.c

```diff
@@ -1429,4 +1454,7 @@ void filter_iterative_manifests(struct list *manifests, int version, struct list
 			*iterative_manifests = list_prepend_data(*iterative_manifests, file);
 		}
 	}
+
+	/* link the iterative manifest with its related full manifest */
+	link_iterative_manifests(*iterative_manifests, *full_manifests);
```

Comment:
I think this function should be moved before the 2nd for loop. This way, the iterative manifest list will not contain any full manifests when linking.